test/test_gpuarray.py

```diff
@@ -468,6 +468,56 @@ def test_arange(self):
         a = gpuarray.arange(12, dtype=np.float32)
         assert (np.arange(12, dtype=np.float32) == a.get()).all()
 
+    
+    @mark_cuda_test
+    def test_stack(input_dims,order):
+
+        shape = (2, 2, 2)[:input_dims]
+        axis = -1 if order == "F" else 0
+
+        from numpy.random import default_rng
+        rng = default_rng()
+        x_in = rng.random(size=shape)
+        y_in = rng.random(size=shape)
+        x_in = x_in if order == "C" else np.asfortranarray(x_in)
+        y_in = y_in if order == "C" else np.asfortranarray(y_in)
+        
+        x_gpu = gpuarray.to_gpu(x_in)
+        y_gpu = gpuarray.to_gpu(y_in)
+
+        np.testing.assert_allclose(gpuarray.stack((x_gpu, y_gpu), axis=axis).get(),
+                                np.stack((x_in, y_in), axis=axis))
+        
+        x_gpu_shape = x_gpu.shape
+        y_gpu_shape = y_gpu.shape
+
+        np.testing.assert_allclose(x_gpu_shape,shape)
+
+        np.testing.assert_allclose(y_gpu_shape,shape)
```

Comment:
What is this supposed to check? `x_in_gpu` is initialized to be an array of shape `shape` and that is again being checked here. I.e. this test is trivial. You probably want to check the equality of shapes of the gpuarray stack result and the numpy array result.

Reply:
Gotcha !! Thanks for the detailed explanation